### PR TITLE
multi: move input to separate package

### PR DESCRIPTION
**Pull request by joostjager**
Isolate input related code to separate package.

This is a follow-up of https://github.com/lightningnetwork/lnd/pull/2198. In that PR a workaround for the weight estimator was introduced to avoid a circular dependency. 

This PR would allow that to be resolved.

### CODE REVIEW

**Comment by joostjager** [comment]
I left `scriptutils_test.go` in the `lnwallet` package, because it would cause a circular dependency if moved. Maybe the test isn't perfectly targeted, but this is something that can be fixed later. Most important would be to get the bulk of the move in asap, so no conflicts can occur anymore.

**Comment by joostjager** [comment]
Coverage dropped 1.4%, but is this real?

**Comment by cfromknecht** [comment]
seems the coverage drop is related to the fact that most of that code was tested indirectly within `lnwallet`. now the `input` package only has direct unit tests for `input.SignDescriptor`. surprised it doesn't do a better job of marking indirect coverage

**Comment by cfromknecht** [comment]
I'm not super concerned about the coverage though, i think the code move is worth it

**Comment by joostjager** [comment]
Looked a bit at the coverage delta. I think it may have to do with our tests being non-deterministic in parts. Maybe the move to another package triggered an internal restructuring that influences the tests and thereby the coverage.

Before:
image

After:
image

**Comment by joostjager** [comment]
Also moved `script_utils` test to input package. Only one test needed to stay inside lnwallet.

File: input/script_utils.go [hunk 1]
```diff
@@ -885,11 +885,11 @@ func HtlcSecondLevelSpend(signer Signer, signDesc *SignDescriptor,
 	return witnessStack, nil
 }
 
-// lockTimeToSequence converts the passed relative locktime to a sequence
+// LockTimeToSequence converts the passed relative locktime to a sequence
 // number in accordance to BIP-68.
 // See: https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki
 //  * (Compatibility)
-func lockTimeToSequence(isSeconds bool, locktime uint32) uint32 {
+func LockTimeToSequence(isSeconds bool, locktime uint32) uint32 {
```
[the remark this answers is not in the document]
Comment (Roasbeef) [reply input/script_utils.go]: Similar comment for `GetStateNumHint`

**Comment by joostjager** [comment]
Moved back the tx related code to `lnwallet`. Isolated other lnwallet moves in separate preparatory commit, to facilitate the review.

PTAL @Roasbeef

File: input/script_utils.go [hunk 2]
```diff
@@ -622,7 +622,7 @@ func receiverHtlcSpendTimeout(signer Signer, signDesc *SignDescriptor,
 	return witnessStack, nil
 }
 
-// createHtlcTimeoutTx creates a transaction that spends the HTLC output on the
+// CreateHtlcTimeoutTx creates a transaction that spends the HTLC output on the
```
[the remark this answers is not in the document]
Comment (cfromknecht) [reply input/script_utils.go]: perhaps we need a `txn_utils` then in `lnwallet`

**Comment by cfromknecht** [comment]
@joostjager would benefit from a rebase, now that #2505 is in

**Comment by joostjager** [comment]
@cdecker coverage -0.002%, so all is good

ptal

**Pull request merged**